domains/pente/src/main/java/io/kaleido/paladin/pente/domain/PenteDomain.java

```diff
@@ -571,25 +575,292 @@ protected CompletableFuture<ToDomain.BuildReceiptResponse> buildReceipt(ToDomain
              }
              if (request.getInfoStatesCount() != 1)
                  throw new IllegalArgumentException("Expected exactly one info state containing the transaction input");
- 
+
              // Recover the input from the signed rawTransaction that is in the "info" state recorded alongside the transaction
              var evmTxn = PenteEVMTransaction.buildFromInput(this, request.getInfoStates(0).getStateDataJson().getBytes(StandardCharsets.UTF_8));
- 
+
              // Do the execution of the transaction again ourselves
              var endorsementLoader = new EndorsementAccountLoader(inputAccounts, readAccounts);
              var execResult = evmTxn.invokeEVM(endorsementLoader);
- 
+
              // Build the full receipt from the result
              var jsonReceipt = evmTxn.buildJSONReceipt(execResult);
- 
-             return CompletableFuture.completedFuture(ToDomain.BuildReceiptResponse.newBuilder().
+
+             return CompletableFuture.completedFuture(BuildReceiptResponse.newBuilder().
                      setReceiptJson(new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(jsonReceipt)).
                      build());
          } catch (Exception e) {
              return CompletableFuture.failedFuture(e);
          }
      }
- 
+
+     void validatePropType(String subSection, JsonABI.Parameter prop, String requiredType) throws IllegalArgumentException {
+         if (!prop.type().equals(requiredType)) {
+             throw new IllegalArgumentException("if you supply a '%s%s' property it must be a '%s' value (type=%s)".formatted(subSection, prop.name(), requiredType, prop.type()));
+         }
+     }
+
+     @Override
+     protected CompletableFuture<InitPrivacyGroupResponse> initPrivacyGroup(InitPrivacyGroupRequest request) {
+
+         try {
+             final var mapper = new ObjectMapper();
+
+             // Process the well-known property names that could be supplied externally
+             var finalABI = JsonABI.newTuple("PentePrivacyGroup", "PentePrivacyGroup", JsonABI.newParameters(
+                     JsonABI.newIndexedParameter("salt", "bytes32"),
+                     JsonABI.newTuple("pente", "PentePrivacyGroupSettings", JsonABI.newParameters(
+                             JsonABI.newParameter("members", "string[]"),
+                             JsonABI.newParameter("evmVersion", "string"),
+                             JsonABI.newParameter("endorsementType", "string"),
+                             JsonABI.newParameter("externalCallsEnabled", "bool")
+                     ))
+             ));
+             var propertiesABI = mapper.readValue(request.getPropertiesAbiJson(), JsonABI.Parameters.class);
```

Comment:
So the implication is that the `propertiesABI` I pass in when initializing the privacy group must be a superset (or at least an overlapping set) with the `finalABI` defined here. At first glance this feels a bit confusing. Should there be two separate ABIs for "well-known properties set by the domain" and "user-defined properties set on group creation"?

Reply:
Also why is "salt" randomly outside the "pente" section?

Reply:
This has all be reconciled into a much simpler system.

Now the `salt` of the genesis state is always generated, and is called `genesisSalt` - to distinguish it from the `salt` that is the previously named (and unchanged) input into each Pente transaction (which is now the `id` of the genesis state / Privacy Group)